### PR TITLE
finish create course and add subject to a course and show list subjec…

### PR DESCRIPTION
**Pull request by vjnhat**
…t on a course

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -1,11 +1,16 @@
 class Course < ActiveRecord::Base
-  has_many :subject_courses
-  has_many :subjects, through: :subject_courses
+  has_many :course_subjects
```
Comment (NguyenTanDuc) [inline app/models/course.rb]: Order

File: app/models/course_subject.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class CourseSubject < ActiveRecord::Base
+  belongs_to :course
+  belongs_to :subject
+
```
Comment (NguyenTanDuc) [inline app/models/course_subject.rb]: remove this line

File: app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -1,12 +1,12 @@
-<% if @user.errors.any? %>
+<% if object.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      The form contains <%= pluralize @user.errors.count, t(:error) %>.
+      The form contains <%= pluralize object.errors.count, "error" %>.
```
Comment (NguyenTanDuc) [inline app/views/shared/_error_messages.html.erb]: I18n error

File: app/views/supervisor/courses/_form.html.erb [hunk 4]
```diff
@@ -1,17 +1,10 @@
...
-  
-  <%= f.label :name %>
-  <%= f.text_field :name, class: 'form-control' %>
-
-  <%= f.label :instruction %>
-  <%= f.text_area :instruction, class: 'form-control' %>
 
-  <%= f.label :start_date %>
-  <%= f.datetime_field :start_date, class: 'form-control' %>
+  <%= f.input :name, :label => "Enter course's name" %>
```
Comment (NguyenTanDuc) [inline app/views/supervisor/courses/_form.html.erb]: remove =>, label: "xxx"

File: app/views/supervisor/courses/show.html.erb [hunk 5]
```diff
@@ -1,4 +1,9 @@
 <%= render @course %>
+
 <%= link_to "delete", supervisor_course_path,
   method: :delete, data: {confirm: t(:confirm_notification)} %>
-<%= link_to "edit", edit_supervisor_course_path %> 
+  
+<%= link_to "edit", edit_supervisor_course_path %>
```
Comment (NguyenTanDuc) [inline app/views/supervisor/courses/show.html.erb]: I18n

File: app/controllers/supervisor/courses_controller.rb [hunk 6]
```diff
@@ -40,7 +40,7 @@ def destroy
   
   private
   def course_params
-    params.require(:course).permit :name, :instruction
+    params.require(:course).permit(:name, :instruction, :subject_ids => [])
```
Comment (NguyenTanDuc) [inline app/controllers/supervisor/courses_controller.rb]: remove ()

File: app/views/supervisor/courses/_form.html.erb [hunk 7]
```diff
@@ -1,17 +1,10 @@
...
+  <%= f.input :instruction, :label => "Enter course's intruction" %>
```
Comment (NguyenTanDuc) [inline app/views/supervisor/courses/_form.html.erb]: same

File: app/controllers/supervisor/courses_controller.rb [hunk 8]
```diff
@@ -40,7 +40,7 @@ def destroy
   
   private
   def course_params
-    params.require(:course).permit :name, :instruction
+    params.require(:course).permit :name, :instruction, :subject_ids => []
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: coding convention

File: app/models/course.rb [hunk 9]
```diff
@@ -1,11 +1,16 @@
...
+  has_many :subjects, through: :course_subjects
+
+  accepts_nested_attributes_for :subjects
+  accepts_nested_attributes_for :course_subjects
   
   validates :name, presence:true, length: {maximum: 50}
   
   before_save :generate_code
 
+  default_scope -> {order(id: :desc)}
```
Comment (dieunb) [inline app/models/course.rb]: remove `()`

File: app/models/subject_course.rb [hunk 10]
```diff
@@ -1,4 +0,0 @@
-class SubjectCourse < ActiveRecord::Base
```
Comment (dieunb) [inline app/models/subject_course.rb]: checkout this file

File: app/models/course.rb [hunk 11]
```diff
@@ -1,11 +1,16 @@
...
+  has_many :subjects, through: :course_subjects
+
+  accepts_nested_attributes_for :subjects
+  accepts_nested_attributes_for :course_subjects
   
   validates :name, presence:true, length: {maximum: 50}
   
   before_save :generate_code
 
+  default_scope -> {order id: :desc}
```
Comment (dieunb) [inline app/models/course.rb]: We don't use it. Remove pls

File: app/models/subject_course.rb (hunk 10 above)
Comment (vjnhat) [inline app/models/subject_course.rb]: I used git rm to removed this file

File: app/views/supervisor/courses/_form.html.erb [hunk 12]
```diff
@@ -1,17 +1,17 @@
...
-  <%= f.datetime_field :start_date, class: 'form-control' %>
 
-  <%= f.label :end_date %>
-  <%= f.datetime_field :end_date, class: 'form-control' %>
-
-  <%= f.submit "Submit", class: "btn btn-primary" %>
-<% end  %>
+  <%= f.input :name %>
+  <%= f.input :instruction %>
+  <%= f.input :start_date, :label => "Start/End dates" do %>
```
Comment (dieunb) [inline app/views/supervisor/courses/_form.html.erb]: coding convention and i18n

File: app/views/supervisor/courses/show.html.erb [hunk 13]
```diff
@@ -1,4 +1,14 @@
...
+
+<%= link_to t(:delete_link), supervisor_course_path,
   method: :delete, data: {confirm: t(:confirm_notification)} %>
-<%= link_to "edit", edit_supervisor_course_path %> 
+  
+<%= link_to t(:edit_link), edit_supervisor_course_path %>
+
+<h3>List subjects</h3>
+<% @course.subjects.each do |s|%>
+  <class="row">
```
Comment (dieunb) [inline app/views/supervisor/courses/show.html.erb]: ???

File: config/routes.rb [hunk 14]
```diff
@@ -13,6 +13,7 @@
     resources :users
     resources :subjects
     resources :courses
+    resources  :subject_courses
```
Comment (dieunb) [inline config/routes.rb]: ???

File: app/views/supervisor/courses/show.html.erb [hunk 15]
```diff
@@ -1,4 +1,13 @@
...
+
+<%= link_to t(:delete_link), supervisor_course_path,
   method: :delete, data: {confirm: t(:confirm_notification)} %>
-<%= link_to "edit", edit_supervisor_course_path %> 
+  
+<%= link_to t(:edit_link), edit_supervisor_course_path %>
+
+<h3>List subjects</h3>
+<% @course.subjects.each do |s| %>
+    <div class="row">
```
Comment (leduytungframgia) [inline app/views/supervisor/courses/show.html.erb]: indent

File: app/controllers/supervisor/courses_controller.rb [hunk 16]
```diff
@@ -26,7 +26,7 @@ def edit
   end
 
   def update
-    @course = Course.find_by params[:id]
+    @course = Course.find params[:id]
     @course.update_attributes course_params
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: `if ... else`

File: app/views/supervisor/courses/show.html.erb [hunk 17]
```diff
@@ -1,4 +1,13 @@
...
+    <%= link_to s.name, supervisor_subject_path(s), class: "col-md-5" %>
```
Comment (dieunb) [inline app/views/supervisor/courses/show.html.erb]: indent

File: lib/templates/erb/scaffold/_form.html.erb [hunk 18]
```diff
@@ -0,0 +1,13 @@
+<%%= simple_form_for(@<%= singular_table_name %>) do |f| %>
```
Comment (dieunb) [inline lib/templates/erb/scaffold/_form.html.erb]: check syntax this file

File: lib/templates/erb/scaffold/_form.html.erb [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+  <%%= f.error_notification %>
+
+  <div class="form-inputs">
+  <%- attributes.each do |attribute| -%>
+    <%%= f.<%= attribute.reference? ? :association : :input %> :<%= attribute.name %> %>
+  <%- end -%>
+  </div>
+
+  <div class="form-actions">
+    <%%= f.button :submit %>
```
Comment (leduytungframgia) [inline lib/templates/erb/scaffold/_form.html.erb]: <%%= ???

File: lib/templates/erb/scaffold/_form.html.erb [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+  </div>
+<%% end %>
```
Comment (leduytungframgia) [inline lib/templates/erb/scaffold/_form.html.erb]: <%% ???

File: app/controllers/supervisor/courses_controller.rb [hunk 21]
```diff
@@ -26,21 +26,24 @@ def edit
...
+    else
+      flash[:danger] =t(:unsuccess)
+      redirect_to suppervisor_courses_path
+    end
   end
 
   def destroy
-    Course.find_by(params[:id]).destroy
+    Course.find (params[:id]).destroy
     flash[:success] = "Course deleted"
```
Comment (NguyenTanDuc) [inline app/controllers/supervisor/courses_controller.rb]: I18n

File: app/controllers/supervisor/courses_controller.rb [hunk 22]
```diff
@@ -26,21 +26,24 @@ def edit
...
-    @course = Course.find_by params[:id]
-    @course.update_attributes course_params
-    flash[:success] = "Course updated"
-    redirect_to supervisor_course_path @course
+    @course = Course.find params[:id]
+    if @course.update_attributes course_params
+      flash[:success] = t(:success)
+      redirect_to supervisor_course_path @course
+    else
+      flash[:danger] =t(:unsuccess)
```
Comment (NguyenTanDuc) [inline app/controllers/supervisor/courses_controller.rb]: space =

File: app/models/course.rb [hunk 23]
```diff
@@ -1,6 +1,9 @@
 class Course < ActiveRecord::Base
-  has_many :subject_courses
-  has_many :subjects, through: :subject_courses
+  has_many :course_subjects, dependent: :destroy
+  has_many :subjects, through: :course_subjects
+
+  accepts_nested_attributes_for :subjects
```
Comment (NguyenTanDuc) [inline app/models/course.rb]: move to end of file

File: app/models/course.rb [hunk 24]
```diff
@@ -1,6 +1,9 @@
...
+  accepts_nested_attributes_for :course_subjects
```
Comment (NguyenTanDuc) [inline app/models/course.rb]: same ^

File: app/views/supervisor/courses/show.html.erb [hunk 25]
```diff
@@ -1,4 +1,13 @@
...
+<%= link_to t(:delete_link), supervisor_course_path,
   method: :delete, data: {confirm: t(:confirm_notification)} %>
-<%= link_to "edit", edit_supervisor_course_path %> 
+  
+<%= link_to t(:edit_link), edit_supervisor_course_path %>
+
+<h3>List subjects</h3>
+<% @course.subjects.each do |s| %>
+  <div class="row">
+  <%= link_to s.name, supervisor_subject_path(s), class: "col-md-5" %>
```
Comment (NguyenTanDuc) [inline app/views/supervisor/courses/show.html.erb]: tab

File: lib/templates/erb/scaffold/_form.html.erb [hunk 26]
```diff
@@ -0,0 +1,13 @@
+<%= simple_form_for(<%= singular_table_name %>) do |f| %>
+  <%= f.error_notification %>
+
+  <div class="form-inputs">
+  <%- attributes.each do |attribute| -%>
```
Comment (NguyenTanDuc) [inline lib/templates/erb/scaffold/_form.html.erb]: tab

**Pull request merged**